plugins/badges-backend/config.d.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface Config {
+  /**
+   * Define all badges.
+   *
+   * The `label` and `message` fields may use templating to support
+   * dynamic content, based on context. Use same syntax as for
+   * javascript template literals, but with `_` instead of `$`,
+   * e.g. `_{context.variable.name}`.
+   *
+   */
+  badges?: {
+    [badgeId: string]: {
+      /**
+       * (Optional) Badge kind.
+       *
+       * Restrict badge usage to a certain kind only.
+       * Useful when using templating for label and/or message if they
+       * use context data only available for a certain kind of badge.
+       *
+       * Available badge kinds:
+       *
+       *   * `entity` The entity data is available as `entity` in the template.
+       *   * `entity_url` The (frontend) URL to view the entity in Backstage.
+       *
+       * Default context for all badges:
+       *
+       *   * `app.title` As configured or defaults to "Backstage".
+       *
+       * @visibility frontend
+       */
+      kind?: 'entity';
+
+      /**
+       * The badge label.
+       *
+       */
+      label: string;
+
+      /**
+       * The badge message.
+       */
+      message: string;
+
+      /**
+       * The message color. Default: `#36BAA2`.
+       */
+      color?: string;
+
+      /**
+       * The label color. Default: `gray`.
+       */
+      labelColor?: string;
+
+      /**
+       * Visual design of the badge. One of: 'plastic', 'flat', 'flat-square',
+       * 'for-the-badge' or 'social'.
+       */
+      style?: 'plastic' | 'flat' | 'flat-square' | 'for-the-badge' | 'social';
+
+      /**
+       * Badge title, used as tooltip text in the markdown code.
+       *
+       * @visibility frontend
+       */
+      title?: string;
+
+      /**
+       * Badge description, used as alt text in the markdown code.
+       * Defaults to badge id.
+       *
+       * @visibility frontend
+       */
+      description?: string;
+
+      /**
+       * Badge link URL, turns badge into a link in the markdown code.
+       *
+       * For `entity` badges, there is a `entity_url` in the context
+       * that could be appropriate to use here.
```

Comment:
This is optional, yes?  reading this, i'm not quite sure what this means, but i assume if i do not include this, then i'll get a link to view my entity WITHOUT needing to specify this?

Reply:
Right. Currently link does not have a default, but for entity badges, it would probably make sense to default it to `entity_url`.

Reply:
Got it; makes sense. if so, I might reword this. I wouldn't say what this will do if you odn't include it, but rather say proactively what it will do. Someething like "overwrites the linked URL to soemthing blah blah...."

Now I'd also counter to ask, what the value of this would be?  i.e., if i understand correctly, an integrator might say ok, this will link to `https://www.google.com?query=mysearch`.  But this will just then e used when they "view the badge" to generate Markdown, and can be JUST as easily changed by the user who grabs the badge - is that right?  if so, in what use cases do you see defining this as YAML standard across an entire instance would be valuable?

Reply:
I'm not 100% sure I understand your concern, but I address them as best I can, see if it makes sense to you.
The `entity_url` will be dynamic, so it points back at the entity for which you're accessing the badges. And yes, the user that grabs the markdown code, can of course change the link at his own discretion. 
I think it is valuable to have as a default, as otherwise you'd have to provide the link to the entity yourself, for every badge.
The integrator are of course free, to choose a different link, or no link at all, if so desired.